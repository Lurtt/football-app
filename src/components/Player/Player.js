import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { margin, fontWeight } from '../mixins'

const PlayerStyle = styled.div`
  color: #ecf0f1;
`

const Position = styled.div`
  ${fontWeight};
`

const Name = styled.p`
  ${margin};
  text-overflow: ellipsis;
  overflow: hidden;
`

export default function Player({ player }) {
  return (
    <PlayerStyle>
      <Position fontWeight={700}>{player.position}</Position>
      <Name margin="0">{player.name}</Name>
    </PlayerStyle>
  )
}

Player.propTypes = {
  player: PropTypes.object.isRequired,
}
