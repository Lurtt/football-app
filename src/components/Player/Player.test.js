import React from 'react'
import renderer from 'react-test-renderer'

import Player from './'

const player = { postion: 'GK', name: 'Darth Vader' }

describe('<Player />', () => {
  test('Should render', () => {
    const result = renderer.create(<Player player={player} />).toJSON()
    expect(result).toMatchSnapshot()
  })
})
