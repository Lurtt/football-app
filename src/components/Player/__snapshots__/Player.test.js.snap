// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> Should render 1`] = `
<div
  className="sc-bdVaJa eMDnEn"
>
  <div
    className="sc-bwzfXH loHIGP"
    fontWeight={700}
  />
  <p
    className="sc-htpNat fFzFcR"
  >
    Darth Vader
  </p>
</div>
`;
